### PR TITLE
WLR simplified build scripts, actions and structure

### PR DESCRIPTION
**Pull request by assambar**
As we bumped WASI SDK and the major runtimes to the latest versions there was a lot of copy-pasting and dealing with mind boggling things in the build scripts.

This PR aims to reduce this for future changes.

 - All target folders follow the `v${MAJOR}.${MINOR}.${BUILD}` format now, as the actual source tag is described in `wlr-env-repo.sh` file. (for example `php/php-8.2.0` would be moved to `php/v8.2.0`). This immensely simplifies GH actions.
 - All build scripts prepare the necessary assets into the `build-output` root now (instead of complex, highly customizable GH Actions doing this for different flavors)
 - GH actions are simplified to dealing with *.{gz,wasm,txt} files instead of complex names. Reusable stuff is extracted in callable workflows
 - All flavor targets are handled by GNU Make (instead of folders with symlinks or copy pasted scripts). Note that all `wlr-build.sh` scripts now have to consider the WLR_BUILD_FLAVOR variable if they want to handle different flavors.
 - Developer notes were moved from README.md to docs/developers.md and updated
 
Note: There is more that could be generalized/simplified but I had to draw the line. The change is already way too big and time consuming for the value it brings.

---
Here is what the release builds for the major runtimes look now
 - PHP - https://github.com/assambar/webassembly-language-runtimes/actions/runs/5544329413
 - Python - https://github.com/assambar/webassembly-language-runtimes/actions/runs/5544330853
 - Ruby - https://github.com/assambar/webassembly-language-runtimes/actions/runs/5544332188

And here are samples for the published releases:
 - PHP - https://github.com/assambar/webassembly-language-runtimes/releases/tag/php%2F8.2.6%2B20230713-c7dee9a
 - Python - https://github.com/assambar/webassembly-language-runtimes/releases/tag/python%2F3.11.4%2B20230713-c7dee9a
 - Ruby - https://github.com/assambar/webassembly-language-runtimes/releases/tag/ruby%2F3.2.2%2B20230713-c7dee9a

### CODE REVIEW

**Review by ereslibre — COMMENTED** [review]
Thank you @assambar. I think this approach is fine, and it will help with maintenance moving forward.

**Review by ereslibre — APPROVED** [review]
This is great, thank you @assambar! I love the direction, and I think is worth getting merged and continuing evolving as we see need.

File: docs/developers.md [hunk 1]
```diff
@@ -1,7 +1,140 @@
 This document is a work in progress.
 
+If it lacks some information that you needed, consider adding it via a PR.
```
Comment (Angelmmiguel) [inline docs/developers.md:3]: This document is amazing! I would remove the first line and add "create an issue" as an alternative for reporting. People may have feedback, but not the time to apply it.

I would suggest changing the 3 lines to:

```md
_Wasm Language Runtimes is evolving constantly so this is a live document. If you find any issue, typo or missing information, feel free to open an issue or create a PR directly_
```

**Review by Angelmmiguel — COMMENTED** [review]
This is a huge and amazing refactor! I believe the build experience is now more consistent and simpler. Congrats on this work! 👏

**Pull request merged**